tests/units/units.cpp

```diff
@@ -1471,3 +1471,327 @@ TEST(Units, compareEquivalentSameSizeButDifferentExponent)
+    model->setName("model");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+    libcellml::UnitsPtr u3 = libcellml::Units::create();
+    libcellml::UnitsPtr u4 = libcellml::Units::create();
+
+    u1->setName("u1"); // Base unit
+    u2->setName("u2");
+    u3->setName("u3");
+    u2->addUnit("u1", 0, 1.0);
+    u2->addUnit(libcellml::Units::StandardUnit::AMPERE, 0, 1.0, 1.0);
+    u3->setName("u3");
+    u3->addUnit("u2", 0, 1.0, 1.0);
+    u4->setName("some_u");
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
+
+    libcellml::ImportSourcePtr import = libcellml::ImportSource::create();
+    import->setUrl("I_am_a_url");
+    import->setModel(model);
+
+    u4->setImportSource(import); // u4 part of a separate model
+    u4->setImportReference("u3");
+
+    EXPECT_TRUE(u4->isImport());
+    EXPECT_FALSE(u4->isBaseUnit());
+}
+
+TEST(Units, isNotBaseUnitsCheckUsingImport)
+{
+    libcellml::ModelPtr model = libcellml::Model::create();
+    model->setName("model");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+    libcellml::UnitsPtr u3 = libcellml::Units::create();
+    libcellml::UnitsPtr u4 = libcellml::Units::create();
+
+    u1->setName("u1"); // Base unit
+    u2->setName("u2");
+    u3->setName("u3");
+    u2->addUnit("u1", 0, 1.0);
+    u2->addUnit(libcellml::Units::StandardUnit::AMPERE, 0, 1.0, 1.0);
+    u3->setName("u3");
+    u3->addUnit("u2", 0, 1.0, 1.0);
+    u4->setName("some_u");
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
+
+    libcellml::ImportSourcePtr import = libcellml::ImportSource::create();
+    import->setUrl("I_am_a_url");
+    import->setModel(model);
+
+    EXPECT_FALSE(u4->isImport());
+    EXPECT_TRUE(u4->isBaseUnit()); // expect true as unit is not declared to be an import
+}
+
+TEST(Units, isBaseUnitsImportModelUnresolved)
+{
+    libcellml::ModelPtr model = libcellml::Model::create();
+    model->setName("model");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+    libcellml::UnitsPtr u3 = libcellml::Units::create();
+    libcellml::UnitsPtr u4 = libcellml::Units::create();
+
+    u1->setName("u1"); // Base unit
+    u2->setName("u2");
+    u3->setName("u3");
+    u2->addUnit("u1", 0, 1.0);
+    u2->addUnit(libcellml::Units::StandardUnit::AMPERE, 0, 1.0, 1.0);
+    u3->setName("u3");
+    u3->addUnit("u2", 0, 1.0, 1.0);
+    u4->setName("some_u");
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
+
+    libcellml::ImportSourcePtr import = libcellml::ImportSource::create();
+    import->setUrl("I_am_a_url");
+
+    u4->setImportSource(import);
+    u4->setImportReference("u3");
+
+    EXPECT_TRUE(u4->isImport());
+    EXPECT_FALSE(u4->isBaseUnit()); // expect false as unit import is unresolved
```

Comment:
Rather than have the comment here it would be better to have the test:
`u4->importSource()->model() == nullptr`.